core/src/test/java/org/elasticsearch/ingest/core/CompoundProcessorTests.java

```diff
@@ -108,4 +114,30 @@ public void testSingleProcessorWithNestedFailures() throws Exception {
         assertThat(processorToFail.getInvokedCounter(), equalTo(1));
         assertThat(lastProcessor.getInvokedCounter(), equalTo(1));
     }
+
+    public void testSetSplitAppend() throws Exception {
+        TemplateService templateService = TestTemplateService.instance();
```

Comment:
It feels to me that this test should move to `SplitProcessorTests`? I see that we need to create two processors, but that is also okay in `SplitProcessorTests`?